src/misc.jl

```diff
@@ -13,23 +14,51 @@ function logmvgamma(p::Int, a::Real)
     return res
 end
 
-# Remainder term after Stirling's approximation to the log-gamma function
-# lstirling(x) = lgamma(x) + x - (x-0.5)*log(x) - 0.5*log2π
-#              = 1/(12x) - 1/(360x^3) + 1/(1260x^5) + ...
-#
-# Asymptotic expansion from:
-#
-#   Temme, N. (1996) Special functions: An introduction to the classical
-#   functions of mathematical physics, Wiley, New York, ISBN: 0-471-11313-1,
-#   Chapter 3.6, pp 61-65.
-#
-# Relative error of approximation is bounded by
-#   (174611/125400 x^-19) / (1/12 x^-1 - 1/360 x^-3)
-# which is < 1/2 ulp for x >= 10.0
-# total numeric error appears to be < 2 ulps
-#
+"""
+    lstirling_asym(x)
+
+Return the remainder term after Stirling's approximation to `lgamma(x)`.
+
+[Stirling's approximation](https://en.wikipedia.org/wiki/Stirling%27s_approximation)
+to the log-gamma (`lgamma`) function also applies to `log(factorial(k))`  through
+the relationship `log(factorial(k)) = lgamma(k + 1)`.  Stirling's approximation is
+
+    lgamma(x) ≈ x*log(x) - x + log(2π/x)/2 = log(x)*(x-0.5) + log2π/2 -x
+
+Thus
+
+    lstirling_asym(x) = lgamma(x) + x - (x-0.5)*log(x) - 0.5*log2π
+                      = 1/(12x) - 1/(360x^3) + 1/(1260x^5) + ...
+
+Why bother with this when `lgamma` can be evaluated directly for most numeric types?
+In some formulas it is advantageous to evaluate the Stirling approximation and
+the remainder to take advantage of explicit cancelation of some terms in
+the Stirling approximation.
+
+Asymptotic expansion from:
+
+   Temme, N. (1996) Special functions: An introduction to the classical
+   functions of mathematical physics, Wiley, New York, ISBN: 0-471-11313-1,
+   Chapter 3.6, pp 61-65.
+
+Relative error of approximation is bounded by
+    (174611/125400 x^-19) / (1/12 x^-1 - 1/360 x^-3)
+which is < 1/2 ulp for x >= 10.0
+total numeric error appears to be < 2 ulps
+"""
+function lstirling_asym end
+
+# For BigInt use the exact calculation of factorial(x-1)
+function lstirling_asym(x::BigInt)
```

Comment:
I think `lgamma` is fine everywhere
```julia
julia> norm([log(factorial(n)) - lgamma(n + 1) for n in big(1):big(500)])
0.0
```
Maybe better to make change the signature to `BigFloat` to match the other versions.

Reply:
Thanks for catching that.